packages/flutter/lib/src/widgets/two_pane.dart

```diff
@@ -0,0 +1,243 @@
+///  hinge. In this case, both [paneProportion] and [direction] are ignored.
+///  * On screens without a hinge, [paneProportion] is used for deciding how
+///  much space each pane uses and [direction] is used for deciding if the 2
+///  panes are laid out horizontally or vertically.
+///
+/// This widget is similar to [Flex], in that it also takes [textDirection] and
+/// [verticalDirection] parameters, which are used for deciding in what order the
+/// panes are laid out (e.g left to right would position pane 1 on the left and
+/// pane 2 on the right).
+///
+/// For narrow screens where you want to display only one pane, you can use [singlePane]
+/// to pick either [TwoPaneSinglePane.pane1] or [TwoPaneSinglePane.pane2].
+///
+/// In addition, this widget also wraps children in [MediaQuery] parents that
+/// makes sense for their side of the screen. For example, let's consider a flip
+/// phone with a notch at the top of the screen. The hinge is horizontal, so the
+/// 2 halves of the screen are one at the top and one at the bottom. In this case
+/// the top pane will have top padding in order to avoid the notch, but the bottom
+/// pane does not have top padding, since it does not have the notch.
+///
+/// See also
+///
+///  * [ui.DisplayFeature] and [MediaQueryData.displayFeatures], to further
+///  understand display features, such as hinge areas
+class TwoPane extends StatelessWidget {
+  /// First pane, which can sit on the left for left to right layouts,
+  /// or at the top for top to bottom layouts.
+  /// If [singlePane] is [TwoPaneSinglePane.pane1], this is the only pane visible.
+  final Widget pane1;
+
+  /// Second pane, which can sit on the right for left to right layouts,
+  /// or at the bottom for top to bottom layouts.
+  /// If [singlePane] is [TwoPaneSinglePane.pane2], this is the only pane visible.
+  final Widget pane2;
+
+  /// Proportion of the screen ocupied by the first pane. The second pane takes
+  /// over the rest of the screen. A value of 0.5 will make the 2 panes equal.
+  /// This property is ignored for displays with a hinge, in which case each
+  /// pane takes over one screen.
+  final double paneProportion;
+
+  /// Same as [Flex.textDirection]
+  final TextDirection? textDirection;
+
+  /// Same as [Flex.verticalDirection]
+  ///
+  /// Defaults to [VerticalDirection.down]
+  final VerticalDirection verticalDirection;
+
+  /// Same as [Flex.direction]
+  ///
+  /// This property is ignored for displays with a hinge, in which case the
+  /// direction is [Axis.horizontal] for vertical hinges and [Axis.vertical] for
+  /// horizontal hinges.
+  ///
+  /// Defaults to [Axis.horizontal]
+  final Axis direction;
+
+  /// Whether to show only one pane and which one, or both. This is useful for
+  /// defining behaviour on narrow devices, where the 2 panes cannot be shown at
+  /// the same time.
+  ///
+  /// Defaults to [TwoPaneSinglePane.both]
+  final TwoPaneSinglePane singlePane;
```

Comment:
Related to comment below on the enum - `panePriority` might be an alternative name here (following the API surface of `TwoPaneView`.